src/main/java/zeenea/connector/dataset/ForeignKey.java

```diff
@@ -30,12 +55,12 @@ public final class ForeignKey {
    * @param builder the builder used to create the ForeignKey instance
    */
   private ForeignKey(Builder builder) {
-    ExceptionUtils.requireNonNullOrEmpty("targetDataset", builder.targetDataset);
-    ExceptionUtils.requireNonNullOrEmpty("sourceFields", builder.sourceFields);
-    ExceptionUtils.requireNonNullOrEmpty("targetFields", builder.targetFields);
     this.targetDataset = builder.targetDataset;
+    this.targetDatasetIdentifier = builder.targetDatasetIdentifier;
     this.sourceFields = List.copyOf(builder.sourceFields);
+    this.sourceFieldIdentifiers = List.copyOf(builder.sourceFieldIdentifiers);
```

Comment:
Attention si `builder.sourceFieldIdentifiers` est `null` ça va lancer une NPE.

Reply:
C'est jamais nul car initialisé dans le builder : 

```
    /** The source fields of the foreign key. */
    @Deprecated(
        since =
            "Deprecated since version 2.1.0, use sourceFieldIdentifiers instead. Scheduled for removal in version 3.0.0.",
        forRemoval = true)
    private List<String> sourceFields = new ArrayList<>();

    /** The list of identifiers for source fields of the foreign key. */
    private List<ItemIdentifier> sourceFieldIdentifiers = new ArrayList<>();
```

Reply:
Après bon, une NPE ici ou là-bas ce n'est pas _forcément_ si différent.

Reply:
Dans le doute, j'ai rajouté un check de non nullité dans le constructeur de la classe `ForeignKey`